From c84a0fda7e2f1add56315a3bf79a5f0146d22f51 Mon Sep 17 00:00:00 2001
From: Jason Zeng <jason.zeng@intel.com>
Date: Mon, 15 Aug 2022 16:51:22 +0800
Subject: [PATCH] Enable SGX virtualization in default config
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: c84a0fda7e2f1add56315a3bf79a5f0146d22f51
Modified-by-SEL: No


Intel inclusion
category: feature
bugzilla: https://gitee.com/openeuler/intel-kernel/issues/I5MJBR
CVE: NA

Intel-SIG: openeuler_defconfig: Enable configs for Intel SGX virtualization

--------------------------------------------

Enable necessary kernel configs for Intel SGX virtualization in openeuler_defconfig.

Reviewed-by: Zhiquan Li <zhiquan1.li@intel.com>
Signed-off-by: Jason Zeng <jason.zeng@intel.com>
Signed-off-by: Aichun Shi <aichun.shi@intel.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/x86/configs/openeuler_defconfig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/x86/configs/openeuler_defconfig b/arch/x86/configs/openeuler_defconfig
index 228a657e4fa8..118e4649fd86 100644
--- a/arch/x86/configs/openeuler_defconfig
+++ b/arch/x86/configs/openeuler_defconfig
@@ -727,6 +727,7 @@ CONFIG_KVM_XFER_TO_GUEST_WORK=y
 CONFIG_VIRTUALIZATION=y
 CONFIG_KVM=m
 CONFIG_KVM_INTEL=m
+CONFIG_X86_SGX_KVM=y
 CONFIG_KVM_AMD=m
 CONFIG_KVM_AMD_SEV=y
 CONFIG_KVM_MMU_AUDIT=y
-- 
2.33.0

